posts/marker-gene-smoothing/index.qmd

```diff
@@ -0,0 +1,714 @@
+
+First, let's take a look at the relative frequency of our T-cell clusters 'Treg', 'T CD8', and 'T CD4'.
+Tregs are supposed to be a rare sub type of T CD4 cells. Yet we have more than 3x as many Tregs called than we do T CD4.  
+
+
+```{r eval=FALSE, message=FALSE}
+#| code-fold: true
+
+bard <- data.table(sem@meta.data)[,.N,by=.(clust)]
+bard[,clust:=factor(clust, levels=bard[order(-N),clust])]
+ggplot(bard[grep("^T",clust)], aes(x=clust, y=N,fill=clust)) + 
+  theme_bw() + 
+  theme(text=element_text(size=16)) + 
+  geom_bar(stat='identity') +
+  scale_fill_manual(values=ctpal, guide = guide_legend(reverse=TRUE)) + 
+  scale_y_continuous(n.breaks=12, labels = scales::comma, name = "# of cells") + 
+  labs(title="# of cells called by T cell subtype") + 
+  geom_text(aes(x=clust, y=N + 200, label=scales::comma(N))) + 
+  coord_flip()
+
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/tcell_calls_barplot.png")
+```
+
+
+Do we really have this many Treg cells in our data?
+
+To gather more evidence toward answering this kind of question, we can calculate the proportion of cells in each cell type expressing a particular marker gene, along with a fold change comparison for each cluster relative to other clusters.
+
+```{r,eval=FALSE,message=FALSE}
+#| code-fold: true
+
+totalcount_norm <- function(sm){
+    libsizes <- Matrix::colSums(sm)
+    scalefactor <- mean(libsizes)
+    libsizes[libsizes==0] <- 1
+    normed <- sm %*% Matrix::Diagonal(x=scalefactor/libsizes)
+    dimnames(normed) <- dimnames(sm)
+    return(normed)
+  }
+
+clusterwise_fold_change_stats <- function(cnts=NULL, normed = NULL, metainfo, clustercol){
+  if(missing(normed)){
+    normed <- totalcount_norm(cnts) 
+  }
+  outl <- list()
+  for(ii in unique(metainfo[[clustercol]])){
+    cells_ii <- metainfo[metainfo[[clustercol]]==ii,cell_ID]
+    cells_iiprime <- metainfo[metainfo[[clustercol]]!=ii,cell_ID]
+    cluster_expr_ii  <- Matrix::rowMeans(normed[,cells_ii,drop=FALSE])
+    cluster_expr_iiprime <- Matrix::rowMeans(normed[,cells_iiprime,drop=FALSE]) 
+    cluster_prop_ii <- Matrix::rowMeans(normed[,cells_ii,drop=FALSE] > 0) 
+    cluster_prop_iiprime <- Matrix::rowMeans(normed[,cells_iiprime,drop=FALSE] > 0) 
+    fctbl <- data.table(cluster=ii
+                        ,cluster_expr = cluster_expr_ii
+                        ,clusterprime_expr = cluster_expr_iiprime
+                        ,gene = names(cluster_expr_ii)
+                        ,cluster_prop = cluster_prop_ii
+                        ,clusterprime_prop = cluster_prop_iiprime
+    )[,typ:=clustercol]
+    fctbl[,fold_change:=cluster_expr / clusterprime_expr]
+    fctbl[,fold_change_prop:=cluster_prop / clusterprime_prop]
+    outl[[paste0(ii)]] <- copy(fctbl) 
+  } 
+  return(fc = rbindlist(outl))
+}
+```
+
+We can see that FOXP3 is expressed in a much larger frequency in Treg cells relative to other T-cell subtypes-- this is good.  
```

Comment:
Throughout, can you italicize gene names in the post? No need to do that in plots.